src/v2/Components/SaleDetailTimer.tsx

```diff
@@ -0,0 +1,101 @@
+import { SaleDetailTimer_sale } from "v2/__generated__/SaleDetailTimer_sale.graphql"
+import { createFragmentContainer, graphql } from "react-relay"
+import * as React from "react"
+import { Flex, Text } from "@artsy/palette"
+import { useTimer } from "v2/Utils/Hooks/useTimer"
+
+export interface SaleDetailTimerProps {
+  sale: SaleDetailTimer_sale
+}
+
+function relativeDaysUntil(parsedDays, suffix) {
+  if (parsedDays === 0) {
+    return `${suffix} Today`
+  } else if (parsedDays === 1) {
+    return `${parsedDays} Day Until ${suffix}`
+  } else {
+    return `${parsedDays} Days Until ${suffix}`
+  }
+}
+
+export const SaleDetailTimer: React.FC<SaleDetailTimerProps> = ({ sale }) => {
+  const endAt = sale?.endAt
+
+  const startAt = sale?.startAt
+
+  const endedAt = sale?.endedAt
+
+  const { hasEnded, time, hasStarted } = useTimer(endAt!, startAt!)
+
+  if (!endAt) {
+    return null
+  }
+
+  const timerCopy = getTimerCopy(time, hasStarted, hasEnded)
+
+  return (
+    <Flex alignItems="center" flexDirection="column">
+      <Text variant="lg" color={"blue100"}>
+        {!!!endedAt && <Text color={timerCopy.color}>{timerCopy.copy}</Text>}
+      </Text>
+    </Flex>
+  )
+}
+
+export const SaleDetailTimerFragmentContainer = createFragmentContainer(
+  SaleDetailTimer,
+  {
+    sale: graphql`
+      fragment SaleDetailTimer_sale on Sale {
+        endAt
+        endedAt
+        startAt
+      }
+    `,
+  }
+)
+
+export const getTimerCopy = (time, hasStarted, hasEnded) => {
+  const { days, hours, minutes, seconds } = time
+
+  const parsedDays = parseInt(days, 10)
+  const parsedHours = parseInt(hours, 10)
+  const parsedMinutes = parseInt(minutes, 10)
+  const parsedSeconds = parseInt(seconds, 10)
+
+  let copy = ""
+  let color = "blue100"
+
+  // Sale has not yet started
+  if (!hasStarted) {
+    copy = relativeDaysUntil(parsedDays, "Bidding Starts")
+  }
+  // Lots are closing
+  else if (hasEnded) {
+    copy = "Lots are closing"
```

Comment:
This feels possibly inaccurate, when the sale is already over we display 'Lots are closing'? I would have expected to maybe not have this rendered at all and it just displays 'Sale Closed ....' w/ the date (or something like that).

Reply:
so 'hasEnded' in this context is the sales endAt which corresponds to the beginning of closing, not the sale's endedAt, that is used on line 39 to not render the text if we have a value for endedAt.

Reply:
Maybe `lotsAreClosing` should be the name of the argument?